lucene/sandbox/src/java/org/apache/lucene/sandbox/search/MultiNormsLeafSimScorer.java

```
@@ -62,6 +67,14 @@
           weightList.add(field.weight);
         }
       }
+
+      if (normsList.isEmpty() == false && normsList.size() != normFields.size()) {
+        throw new IllegalArgumentException(
+            getClass().getSimpleName()
+                + " requires norms to be consistent across fields: some fields cannot"
+                + " have norms enabled, while others have norms disabled");
```

It's kind of an edge case but should we also throw the exception if scoring is not needed ?

---

Properly throwing an error would require a more invasive change, since we rewrite the query very early if scores aren't needed: https://github.com/apache/lucene/blob/9942d59f0dc92db28d1dd79bf321cd2a05cdb6d1/lucene/sandbox/src/java/org/apache/lucene/sandbox/search/CombinedFieldQuery.java#L262-L265. However I could refactor this part in `MultiNormsLeafSimScorer` just to be clearer/ more solid?

This also made me notice -- I'm using the fact that `LeafReader#getNormsValues` is `null` to detect that norms were disabled. This seems accurate but I couldn't confirm it's really part of the method contract. I wonder if checking field infos is better.

---

Doing more research, I don't think it's possible to detect if norms are disabled upfront: `LeafReader#getNormsValues` could be `null` a field uses norms but happens to be missing from a segment. I opted to just document this requirement (as we do for the fact norms must be additive and encoded in a certain way). If the calling application has its own notion of schema, it can enforce the norms requirements upfront.